File: SearchingTechniques/binarySearch/BinarySearch.java
package SearchingTechniques.binarySearch;

public class BinarySearch {

    public int search(int[] arr, int key) {
        int start = 0;
        int end = arr.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (key < arr[mid]) {
                end = mid - 1;
            } else if (key > arr[mid]) {
                start = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;

    }

    public static void main(String[] args) {
        BinarySearch obj = new BinarySearch();
        int[] arr = { 2, 3, 6, 7, 8, 9 };
        int key = 17;
        int result = obj.search(arr, key);
        if (result == -1) {
            System.out.println("the key " + key + " not found!");
        } else {
            System.out.println("The Key " + key + " found at the location " + result);
        }
    }
}

File: JAVA21/begining/DSA in C/WEEK2/BinarySearch.c
#include<stdio.h>
int binarysearch(int arr[],int target,int low,int high){
    while(low<=high){
        int mid = low+(high-low)/2;
        if(arr[mid]==target){
            return mid;
        }else if(arr[mid]<target){
            low = mid+1;
        }
        else{
            high = mid-1;
        }
    }
    return -1;
}
int main(){
    int n,target;
    printf("enter the size of the array:");
    scanf("%d",&n);
    int arr[n];
    printf("insert the numbers in the array:");
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    printf("enter the element to search:");
    scanf("%d",&target);
    int result = binarysearch(arr,target,0,n-1);
    if (result!=-1){
        printf("element %d found at the location %d",target,result+1);
    }else{
        printf("the element %d not found in the list!",target);
    }
    return 0;

}